src/main/java/de/tum/in/www1/artemis/web/rest/TextExerciseResource.java

```diff
@@ -76,6 +81,15 @@ public TextExerciseResource(TextExerciseRepository textExerciseRepository,
         if (textExercise.getId() != null) {
             throw new BadRequestAlertException("A new textExercise cannot already have an ID", ENTITY_NAME, "idexists");
         }
+
+        if (textExercise.getTitle() == null) {
+            throw new BadRequestAlertException("A new textExercise needs a title", ENTITY_NAME, "missingtitle");
+        }
+
+        if (textExercise.getDueDate() == null) {
```

Comment:
We should also allow instructor to create text exercises without due date.

Reply:
Think about how to handle and display such a case in the tutor exercise and tutor course dashboard

Reply:
Done! They still appear in the tutor course dashboard, but after the one with a valid assessmentDueDate